### Задча
Проанализируйте результаты эксперимента и напишите свои рекомендации менеджеру.

Mobile Games AB Testing with Cookie Cats

In [4]:
import pandas as pd

# Загрузим данные из CSV-файла
data = pd.read_csv('C:/Users/clic/OneDrive/Рабочий стол/Stud/A.B.Test/Sem9DZ/А_B_cookie_cats2.csv')

print(data)

        userid  version  sum_gamerounds  retention_1  retention_7
0          116  gate_30               3            0            0
1          337  gate_30              38            1            0
2          377  gate_40             165            1            0
3          483  gate_40               1            0            0
4          488  gate_40             179            1            1
...        ...      ...             ...          ...          ...
90184  9999441  gate_40              97            1            0
90185  9999479  gate_40              30            0            0
90186  9999710  gate_30              28            1            0
90187  9999768  gate_40              51            1            0
90188  9999861  gate_40              16            0            0

[90189 rows x 5 columns]


1. Распределение версий:

* В тестировании использовались две версии игры: "gate_30" и "gate_40".
* Важно убедиться, что распределение пользователей между версиями было случайным и сбалансированным, чтобы избежать искажения результатов. 
* Менеджер должен проверить, что пользователи были распределены равномерно.

In [5]:
import pandas as pd

data = pd.read_csv('C:/Users/clic/OneDrive/Рабочий стол/Stud/A.B.Test/Sem9DZ/А_B_cookie_cats2.csv')

# Посмотрим на распределение версий
version_distribution = data['version'].value_counts()
print(version_distribution)


version
gate_40    45489
gate_30    44700
Name: count, dtype: int64


* Версия "gate_40" имеет количество пользователей равное 45489.
* Версия "gate_30" имеет количество пользователей равное 44700.
Распределение пользователей между версиями примерно равное, что говорит о том, что тестирование было проведено сбалансированно и случайным образом. Это важно для обеспечения достоверности результатов эксперимента.

2. Участие пользователей (sum_gamerounds):

"sum_gamerounds" представляет собой общее количество сыгранных пользователем раундов в игре за тестовый период.
* Следует сравнить среднее или медианное количество раундов между версиями, чтобы определить, есть ли значительная разница в участии пользователей.
* Если среднее количество раундов существенно различается, это может указывать на более захватывающую или увлекательную версию игры.

In [6]:
import pandas as pd

data = pd.read_csv('C:/Users/clic/OneDrive/Рабочий стол/Stud/A.B.Test/Sem9DZ/А_B_cookie_cats2.csv')

# Сгруппируем данные по версии и посчитаем среднее количество раундов
average_gamerounds_by_version = data.groupby('version')['sum_gamerounds'].mean()
print(average_gamerounds_by_version)


version
gate_30    52.456264
gate_40    51.298776
Name: sum_gamerounds, dtype: float64


* Версия "gate_30" имеет среднее количество сыгранных раундов, приблизительно равное 52.46.
* Версия "gate_40" имеет среднее количество сыгранных раундов, приблизительно равное 51.30.
Среднее количество раундов для версии "gate_30" незначительно выше, чем для версии "gate_40", но различия не кажутся значительными.

3. Показатели удержания:

* Retention_1 представляет процент пользователей, которые вернулись в игру на следующий день после первой установки 
* Retention_7 отражает процент пользователей, которые вернулись через семь дней после первой установки.
Менеджеру следует сравнить показатели удержания между двумя версиями.
Если одна из версий показывает более высокие показатели удержания (как на 1-й, так и на 7-й день), это указывает на то, что эта версия более успешна в удержании игроков.

In [7]:
import pandas as pd

data = pd.read_csv('C:/Users/clic/OneDrive/Рабочий стол/Stud/A.B.Test/Sem9DZ/А_B_cookie_cats2.csv')

# Посчитаем показатель удержания на 1 и 7 день для каждой версии
retention_1_by_version = data.groupby('version')['retention_1'].mean()
retention_7_by_version = data.groupby('version')['retention_7'].mean()

print("Retention_1 by version:")
print(retention_1_by_version)

print("Retention_7 by version:")
print(retention_7_by_version)


Retention_1 by version:
version
gate_30    0.448188
gate_40    0.442283
Name: retention_1, dtype: float64
Retention_7 by version:
version
gate_30    0.190201
gate_40    0.182000
Name: retention_7, dtype: float64


Оба показателя удержания (Retention_1 и Retention_7) для версии "gate_30" немного выше, чем для версии "gate_40". Однако различия в показателях незначительные.


В целом, обе версии демонстрируют относительно хорошие показатели удержания, и выбор предпочтительной версии может зависеть от других факторов. Рекомендуется провести дополнительный анализ и учесть другие факторы перед принятием окончательного решения о том, какая версия следует предпочесть.

4. Анализ низкого удержания:


In [8]:
import pandas as pd

data = pd.read_csv('C:/Users/clic/OneDrive/Рабочий стол/Stud/A.B.Test/Sem9DZ/А_B_cookie_cats2.csv')

# Выберем пользователей с низким уровнем удержания (retention_1 = 0) и проанализируем причины
low_retention_users = data[data['retention_1'] == 0]
print(low_retention_users)
# Выведем дополнительные данные для анализа низкого удержания
print(low_retention_users.describe())


        userid  version  sum_gamerounds  retention_1  retention_7
0          116  gate_30               3            0            0
3          483  gate_40               1            0            0
6         1066  gate_30               0            0            0
7         1444  gate_40               2            0            0
10        1842  gate_40               3            0            1
...        ...      ...             ...          ...          ...
90178  9998555  gate_40              32            0            0
90180  9998623  gate_30               7            0            0
90183  9999349  gate_30              10            0            0
90185  9999479  gate_40              30            0            0
90188  9999861  gate_40              16            0            0

[50036 rows x 5 columns]
             userid  sum_gamerounds  retention_1   retention_7
count  5.003600e+04    50036.000000      50036.0  50036.000000
mean   5.005118e+06       17.345651          0.0      0.

* Количество пользователей с низким уровнем удержания (Retention_1 = 0) составляет 50036 человек.

* Среднее количество сыгранных раундов (sum_gamerounds) для этих пользователей составляет приблизительно 17.35.

* Эти пользователи не возвращаются в игру на следующий день после первой установки, что говорит о низком уровне удержания в начальный период игры.

* Показатель удержания через семь дней (Retention_7) для этих пользователей также невысок (среднее значение около 0.072), что указывает на небольшое количество пользователей, которые возвращаются в игру через семь дней после первой установки.

Для увеличения удержания и привлечения пользователей к продолжительной игре менеджеру можно рекомендовать следующее:

1. Изучить первоначальный опыт новых пользователей.

2. Увеличить внимание к новым пользователям.
3. Улучшить отзывы и обратная связь.

5. Производительность версий:

Сравнить общую производительность каждой версии на основе сочетания участия пользователей и удержания.
Определить, у какой версии выше показатели удержания и участия пользователей в целом.

In [9]:
import pandas as pd

data = pd.read_csv('C:/Users/clic/OneDrive/Рабочий стол/Stud/A.B.Test/Sem9DZ/А_B_cookie_cats2.csv')

# Посчитаем общую производительность каждой версии на основе участия пользователей и удержания
performance_by_version = data.groupby('version').agg(
    average_gamerounds=('sum_gamerounds', 'mean'),
    retention_1=('retention_1', 'mean'),
    retention_7=('retention_7', 'mean')
)

print(performance_by_version)


         average_gamerounds  retention_1  retention_7
version                                              
gate_30           52.456264     0.448188     0.190201
gate_40           51.298776     0.442283     0.182000


Общая производительность обеих версий схожа, и отличия в среднем количестве сыгранных раундов и показателях удержания между "gate_30" и "gate_40" незначительны. При таком сценарии выбор предпочтительной версии может зависеть от других факторов, таких как обратная связь пользователей, оценки популярности функций и возможностей каждой версии.

6. Статистическая значимость: 
* Различия в среднем количестве сыгранных раундов между версиями "gate_30" и "gate_40" являются статистически значимыми, что подтверждает разницу в участии пользователей.

In [11]:
import pandas as pd
from scipy import stats

data = pd.read_csv('C:/Users/clic/OneDrive/Рабочий стол/Stud/A.B.Test/Sem9DZ/А_B_cookie_cats2.csv')

# Выполним t-тест для сравнения средних значений sum_gamerounds между версиями
gate_30_gamerounds = data[data['version'] == 'gate_30']['sum_gamerounds']
gate_40_gamerounds = data[data['version'] == 'gate_40']['sum_gamerounds']
t_statistic, p_value = stats.ttest_ind(gate_30_gamerounds, gate_40_gamerounds)
print("\nРезультаты статистического теста (sum_gamerounds):")
print("t-статистика:", t_statistic)
print("p-значение:", p_value)


Результаты статистического теста (sum_gamerounds):
t-статистика: 0.8910426211362967
p-значение: 0.37290868247405207


Значение t-статистики: 0.891
Значение p-значения: 0.373
Вывод:
P-значение (0.373) является значительно большим, чем уровень значимости (например, 0.05). Это означает, что мы не можем отвергнуть нулевую гипотезу, согласно которой нет статистически значимых различий в средних значениях sum_gamerounds между версиями "gate_30" и "gate_40".

На основе проведенного теста, нет оснований считать, что одна версия игры привлекает значительно больше пользователей, чем другая. Обе версии игры, имеют схожее влияние на активность пользователей в игре, и различия в средних значениях sum_gamerounds между версиями могут быть случайными.

### Рекомендации менеджеру

1. Распределение пользователей между версиями игры "gate_30" и "gate_40" примерно равное. Это говорит о том, что тестирование было проведено сбалансированно и случайным образом, что важно для обеспечения надежности результатов эксперимента.

2. Среднее количество сыгранных раундов  между версиями "gate_30" и "gate_40" не имеет значительной разницы. Это означает, что обе версии игры примерно одинаково привлекают и удерживают пользователей, и нет оснований полагать, что одна версия значительно лучше другой.

3. Показатели удержания для обеих версий игры также не сильно различаются. Обе версии демонстрируют относительно хорошие показатели удержания, и выбор предпочтительной версии может зависеть от других факторов.

4. Анализ пользователей с низким уровнем удержания (Retention_1 = 0) показывает, что есть некоторое количество пользователей, которые не возвращаются в игру на следующий день после первой установки. Для увеличения удержания и привлечения пользователей к продолжительной игре можно рекомендовать обратить внимание на первоначальный опыт новых пользователей, улучшить обратную связь и отзывы, а также увеличить внимание к новым пользователям.

5. Общая производительность обеих версий игры схожа, и различия в показателях удержания и участия пользователей между "gate_30" и "gate_40" незначительны. Рекомендуется учесть другие факторы, такие как обратная связь пользователей, оценки популярности функций и возможностей каждой версии, перед принятием окончательного решения о предпочтительной версии.

6. Статистический тест показал нам, что различия в средних значениях sum_gamerounds между версиями "gate_30" и "gate_40" являются статистически незначимыми. Это подтверждает, что различия в участии пользователей между версиями могут быть случайными.

Обе версии игры Cookie Cats демонстрируют схожую производительность, и выбор предпочтительной версии следует основывать на других факторах, таких как обратная связь пользователей, стоимость разработки и поддержки каждой версии, и ожидаемые долгосрочные результаты. Также, для улучшения удержания пользователей, стоит обратить внимание на первоначальный опыт новых пользователей и возможные причины низкого удержания.